nemo/collections/nlp/models/language_modeling/megatron_ptune_t5_model.py

```diff
@@ -0,0 +1,494 @@
+            tensor_model_parallel_size=cfg.get('tensor_model_parallel_size', 1),
+            seed=cfg.get('seed', 1234),
+        )
+
+        # shared params for dataset and data loaders
+        # tokenizer needs to get initialized before the super.__init__()
+        # as dataloaders and datasets need it to process the data
+
+        self.megatron_amp_o2 = cfg.get('megatron_amp_O2', False)
+        # TODO: Fix this once apex patches FusedScaledMaskedSoftmax.
+        # This is a workaround for the fact that `masked_softmax_fusion` has issues with certain input sizes that may be present while finetuning.
+        t5_cfg = MegatronT5Model.restore_from(
+            self.register_artifact('language_model.nemo_file', cfg.language_model.get('nemo_file', None)),
+            trainer=trainer,
+            return_config=True,
+        )
+        OmegaConf.set_struct(t5_cfg, True)
+        with open_dict(t5_cfg):
+            t5_cfg.masked_softmax_fusion = False
+            t5_cfg.megatron_amp_O2 = self.megatron_amp_o2
+
+        self.model = MegatronT5Model.restore_from(
+            self.register_artifact('language_model.nemo_file', cfg.language_model.get('nemo_file', None)),
+            trainer=trainer,
+            override_config_path=t5_cfg,
+        )
+
+        # self.model = MegatronT5Model.restore_from(
+        #     self.register_artifact('language_model.nemo_file', cfg.language_model.get('nemo_file', None)),
+        #     trainer=trainer)
+
+        self.tokenizer = self.model.tokenizer
+
+        self.float_type = self.model.enc_dec_model.enc_dec_model.encoder.model.layers[0].dtype
+
+        if not cfg.use_lm_finetune:
+            self.model.freeze()
+
+        hidden_size = self.model.cfg.hidden_size
+
+        # register the file containing the labels into the artifacts to get stored in the '.nemo' file later
+        self.word_embeddings = self.model.enc_dec_model.encoder_embedding.word_embeddings
+        self.position_embeddings = self.model.enc_dec_model.encoder_embedding.position_embeddings
+
+        # self.vocab = self.tokenizer.tokenizer.get_vocab()
+
+        self.template = cfg.prompt_encoder.template
+
+        self.prompt_encoder = PromptEncoder(
+            template=cfg.prompt_encoder.template,
+            hidden_size=hidden_size,
+            lstm_dropout=cfg.prompt_encoder.dropout,
+            num_layers=cfg.prompt_encoder.num_layers,
+        )
+
+        # load prompt encoder
+        self.hidden_size = hidden_size
+        self.tokenizer.add_special_tokens([cfg.pseudo_token])
+
+        self.pseudo_token_id = self.tokenizer.special_token_to_id[cfg.pseudo_token]
+        self.pad_token_id = self.tokenizer.pad_id if self.tokenizer.pad_id is not None else self.tokenizer.unk_id
+        self.spell_length = sum(self.template)
+        self._reduced_loss_buffer = []
+        self.decoder_seq_length = cfg.get('decoder_seq_length', 10)
+
+    def embed_input(self, queries, enc_taskname):
```

Comment:
Can you change `queries`/`enc_taskname` to a more meaningful name? perhaps add `_id`?

Reply:
fixed